src/security/core/src/dds_security_serialize.c

```diff
@@ -49,6 +49,7 @@
 #define PID_TRANSPORT_PRIORITY                  0x49u
 #define PID_PROTOCOL_VERSION                    0x15u
 #define PID_VENDORID                            0x16u
+#define PID_PRODUCT_VERSION                     0x8000u
```

Comment:
As the product version is only informational now, I don't think anything is to be gained by also burdening the serializer used for the DDS Security with it. The clear text SPDP message will be good enough for the purpose.

Note that we shouldn't even have a separate "DDS Security" serializer. It only exists because the in-memory representation in the spec'd plug-in interface is different, and the dice landed the wrong way when a choice had to be made between adding a second serializer or having only a single serializer and converting between in-memory representations. Given the security sensitivity of (in particular de-)serializers, we should get rid of the second one.

In short, I wouldn't add it here.